programs/drift/src/math/margin.rs

```diff
@@ -275,7 +275,15 @@ pub fn calculate_margin_requirement_and_total_collateral_and_liability_info(
 
         if spot_market.market_index == 0 {
             let token_amount = spot_position.get_token_amount(&spot_market)?;
-
+            if token_amount == 0 {
+                validate!(
+                    spot_position.scaled_balance == 0,
```

Comment:
since `get_token_amout` rounds down, this could be a small value and the token amount could be zero